app_server/app/adapters/socket/Room.js

```diff
@@ -75,4 +75,21 @@ Class.removeAllClients = function() {
   this.__clients = {};
 };
 
+Class.getNumberOfUsers = function() {
+  let users = {};
+  let count = 0;
+  for (var socketId in this.getClients()) {
+    let userId = this.getClient(socketId).getUserId(); 
+    if (!users[userId]) {
+      users[userId] = true;
+      count++;
+    }
+  }
+  return count;
```

Comment:
Actually, what type is `userId`?
If it's a string or some basic type, shouldn't it be possible to `map` it into an array and toss it into a `new Set` before returning its `length`?